### PR TITLE
feat(function-tree): Use primitives to identify types of execution

### PR DESCRIPTION
**Pull request by christianalfoni**
After a few iterations now this is the result:

```js
import {sequence, parallel} from 'cerebral' // or function-tree
```

## Sequence
So sequence is normal execution, one after the other:

```js
sequence(
  funcA () {},
  funcB () {}
)
```

Though instead of calling the `sequence` function, you can just use a plain array. That means `[]` is converted to `sequence()` inside function-tree.

```js
[
  funcA () {},
  funcB () {}
]
```

The reason you would want to use `sequence` is to be more explicit and/or name your sequence:

```js
sequence(
  'My sequence',
  funcA () {},
  funcB () {}
)
```

When naming the sequence it is picked up by the debugger.

## Parallel
Well, it is parallel execution. Meaning that "array inside array" is gone, you have to use this function to express parallel execution.

```js
parallel(
  'I can optionally name this as well',
  funcA () {},
  funcB () {}
)
```

When using `parallel` any group of actions (sequence) will be run as "one action". Meaning that:

```js
parallel(
  funcAAsync () {},
  [
      funcBAsync () {},
      funcC () {}
  ]
)
```

`funcC` would run after `funcBAsync` is done, it would not get merged in as a parallel action. This is a lot more predictable behaviour and in the debugger you will see that there is a `parallel` group of actions with a `sequence` group of actions inside it.

Here is a simple screenshot of how debugger looks with the identified grouping:

screen shot 2017-02-25 at 12 52 29

The really great bonus now is that compositions are identified and visualized in debugger. Because you typically import arrays from other chains etc. When not using spread these become their own `sequence` group of actions indicated in the debugger. And if you named them... even better :)

### CODE REVIEW

**Comment by henri-hulski** [comment]
Is it now possible to use a single action without array?

File: packages/cerebral/src/operators/all.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export {all as default} from 'function-tree'
```
Comment (henri-hulski) [inline packages/cerebral/src/operators/all.js]: We have still an all operator.

File: packages/cerebral/src/operators/debounce.test.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env mocha */
 import Controller from '../Controller'
 import assert from 'assert'
-import {debounce} from './'
+import {all, debounce} from './'
```
Comment (henri-hulski) [inline packages/cerebral/src/operators/debounce.test.js]: The all operator still used.

File: packages/cerebral/src/operators/wait.test.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env mocha */
 import Controller from '../Controller'
 import assert from 'assert'
-import {wait} from './'
+import {all, wait} from './'
```
Comment (henri-hulski) [inline packages/cerebral/src/operators/wait.test.js]: Here also.

File: packages/cerebral/src/operators/all.js (hunk 1 above)
Comment (christianalfoni) [inline packages/cerebral/src/operators/all.js]: Thanks, fixed :)

File: packages/cerebral/src/operators/debounce.test.js (hunk 2 above)
Comment (christianalfoni) [inline packages/cerebral/src/operators/debounce.test.js]: Thanks, fixed

File: packages/cerebral/src/operators/wait.test.js (hunk 3 above)
Comment (christianalfoni) [inline packages/cerebral/src/operators/wait.test.js]: Thanks, fixed

**Comment by christianalfoni** [comment]
Okay, I just need to update function tree debugger to support these changes and it is ready to go :)

**Comment by Guria** [comment]
I was away this days and not followed any discussions. I am positive with this change, though wondering about breaking for current alpha users.

**Comment by christianalfoni** [comment]
@Guria Welcome back :D The only thing that breaks with this change is parallel execution. You have to identify where you do array in array and use `parallel` instead. Due to the nature of the change I could not see how to deprecate this.

If you use `spread` operator, that is still perfectly okay, you will just not get the "sequence" indication in the debugger, as it will think of it as "one sequence".

The static tree also look different now, so need to release new debugger at the same time as this. I suggest we release it on the next hangout to inform people and also make sure we shout out quite a bit about it the days before. So official release is next thursday.

Or is there a better way? :)

**Pull request merged**